notebooks/tutorials/overland_flow/overland_flow_erosion/ol_flow_erosion_components.ipynb

```diff
@@ -0,0 +1,489 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## `DepthSlopeProductErosion`\n",
+    "\n",
+    "This component represents the rate of erosion, $E$, by surface water flow as:\n",
+    "\n",
+    "$$E = k_e (\\tau^a - \\tau_c^a)$$\n",
+    "\n",
+    "where $k_e$ is an erodibility coefficient (with dimensions of velocity per stress$^a$), $\\tau$ is bed shear stress, $\\tau_c$ is a minimum bed shear stress for any erosion to occur, and $a$ is a parameter that is commonly treated as unity.\n",
+    "\n",
+    "For steady, uniform flow, \n",
+    "\n",
+    "$$\\tau = \\rho g H S$$,\n",
+    "\n",
+    "with $\\rho$ being fluid density, $g$ gravitational acceleration, $H$ local water depth, and $S$ the (postive-downhill) slope gradient (an approximation of the sine of the slope angle).\n",
+    "\n",
+    "The component uses a user-supplied slope field (at nodes) together with the water-depth field `surface_water__depth` to calculate $\\tau$, and then the above equation to calculate $E$. The component will then modify the `topographic__elevation` field accordingly. If the user wishes to apply material uplift relative to baselevel, an `uplift_rate` parameter can be passed on initialization.\n",
+    "\n",
+    "Here are the header and constructor docstrings:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from landlab.components import DepthSlopeProductErosion\n",
+    "\n",
+    "print(DepthSlopeProductErosion.__doc__)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(DepthSlopeProductErosion.__init__.__doc__)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Example\n",
+    "\n",
+    "In this example, we load the topography of a small drainage basin, calculate a water-depth field by running overland flow over the topography using the `KinwaveImplicitOverlandFlow` component, and then calculating the resulting erosion.\n",
+    "\n",
+    "Note that in order to accomplish this, we need to identify which variable we wish to use for slope gradient. This is not quite as simple as it may sound. An easy way to define slope is as the slope between two adjacent grid nodes. But using this approach means that slope is defined on the grid *links* rathter than *nodes*. To calculate slope magnitude at *nodes*, we'll define a little function below that uses Landlab's `calc_grid_at_link` method to calculate gradients at grid links, then use the `map_link_vector_components_to_node` method to calculate the $x$ and $y$ vector components at each node. With that in hand, we just use the Pythagorean theorem to find the slope magnitude from its vector components.\n",
```

Comment:
"... we'll define a little function below that uses Landlab's `calc_grad*_at_link` method..."

Reply:
fixed